### ISSUE TITLE
500 error when upload file greater than 1M via rpc.php

### ISSUE THREAD

**Issue by L3o-pold** [crazy-max/docker-rtorrent-rutorrent#245]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/crazy-max/docker-rtorrent-rutorrent/blob/master/.github/SUPPORT.md)

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When uploading a torrent greater than 1M on `/plugins/rpc/rpc.php` a 500 HTTP header occurs. I found no interesting error log to investigate. I already increased `XMLRPC_SIZE_LIMIT` to `16M` and `MEMORY_LIMIT` to `2048M` without success.

```
rutorrent-6db6c7bb8c-slkzs rutorrent 2023/05/22 15:37:56 [warn] 552#552: *681 a client request body is buffered to a temporary file /tmp/nginx/8/00/0000000008, client: 10.0.0.128, server: , request: "POST /plugins/rpc/rpc.php HTTP/1.1", host: "192.168.1.200:31705"
rutorrent-6db6c7bb8c-slkzs rutorrent 10.0.0.128 - - [22/May/2023:15:37:56 +0000] "POST /plugins/rpc/rpc.php HTTP/1.1" 500 48 "-" "Sonarr/4.0.0.500 (alpine 3.16.5)"
rutorrent-6db6c7bb8c-slkzs rutorrent 2023/05/22 15:37:59 [info] 552#552: *684 client closed connection while waiting for request, client: 10.0.0.128, server: 0.0.0.0:8080
rutorrent-6db6c7bb8c-slkzs rutorrent 2023/05/22 15:37:59 [info] 552#552: *683 client closed connection while waiting for request, client: 10.0.0.128, server: 0.0.0.0:8080
```

### Expected behaviour

Should return an 200 http

### Actual behaviour

500 error code

### Steps to reproduce

1. Upload a large torrent to  `/plugins/rpc/rpc.php`
2. see the 500 error

### Docker info

```yaml
Kubernetes
```

### Version

* Docker compose version (type `docker-compose --version`):
* System info (type `uname -a`):

### Docker compose

```yaml
Kubernetes
```

### Container logs

```
rutorrent-6db6c7bb8c-slkzs rutorrent 2023/05/22 15:37:56 [warn] 552#552: *681 a client request body is buffered to a temporary file /tmp/nginx/8/00/0000000008, client: 10.0.0.128, server: , request: "POST /plugins/rpc/rpc.php HTTP/1.1", host: "192.168.1.200:31705"
rutorrent-6db6c7bb8c-slkzs rutorrent 10.0.0.128 - - [22/May/2023:15:37:56 +0000] "POST /plugins/rpc/rpc.php HTTP/1.1" 500 48 "-" "Sonarr/4.0.0.500 (alpine 3.16.5)"
rutorrent-6db6c7bb8c-slkzs rutorrent 2023/05/22 15:37:59 [info] 552#552: *684 client closed connection while waiting for request, client: 10.0.0.128, server: 0.0.0.0:8080
rutorrent-6db6c7bb8c-slkzs rutorrent 2023/05/22 15:37:59 [info] 552#552: *683 client closed connection while waiting for request, client: 10.0.0.128, server: 0.0.0.0:8080
```

### Additional info

Using 4.1.5-0.9.8-0.13.8 docker image

**Comment by f0oster** [comment]
Seeing the same behavior with the latest image. 

Attempting a 1,715KB .torrent file upload via RPC.php triggers HTTP 500 status code. In a similar boat with logs - nothing useful logged.

**Comment by stickz** [comment]
Could you try setting `RU_REMOVE_CORE_PLUGINS` to `rpc`? This will use `httprpc` instead.

**Comment by L3o-pold** [comment]
@stickz it's not working `10.0.0.128 - - [23/May/2023:07:19:37 +0000] "POST /plugins/rpc/rpc.php HTTP/1.1" 404 146 "-" "Sonarr/4.0.0.500 (alpine 3.16.5)"`

**Comment by f0oster** [comment]
> @f0oster for me it's working from the rutorrent interface. Do you have an nginx ingress in front?

Good point... I do have an nginx reverse proxy configured which I was likely using during testing, whoops.

I'll test and confirm shortly.

**Comment by L3o-pold** [comment]
@stickz I don't think Sonarr works with `httprpc`.

**Comment by stickz** [comment]
> @stickz I don't think Sonarr works with `httprpc`.

It does. You just need to change the url base path. ruTorrent will accept the `XML` response from Sonarr through `httprpc`.
https://forums.sonarr.tv/t/rtorrent-support/3644/22

**Comment by stickz** [comment]
> rutorrent returns a 200 HTTP code, but Sonarr fails

You can try to talk directly to rTorrent on Sonarr through the URL path `RPC2` or `/RPC2` on port 8000. Then use `httprpc` for ruTorrent. This is a better configuration because ruTorrent won't be required to function properly. This docker image exposes that.

**Comment by stickz** [comment]
When switching Sonarr to `/RPC2` you received the same error, but a more detailed a 502 instead of 500 error code. I can't be of any further assistance. I would recommend to continue using `/RPC2` because it takes ruTorrent out of the picture.

**Comment by FloydKulis** [comment]
I have a similar problem in Radarr. When trying to download a torrent file, I receive the following message:
HTTP request failed: [413:RequestEntityTooLarge] [POST] at [http://192.168.50.176:8000/RPC2]
Adding a torrent directly via rutorrent works without any problems.
Does anyone know any solution?
Can anyone tell me where I can add file limit parameters for rpc?

**Comment by phene** [comment]
I had an nginx front-end and fixed this issue by adjusting [client_max_body_size](https://www.slingacademy.com/article/nginx-client_max_body_size-explained-with-examples/)
